src/components/Actions.jsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import { patchArticleByID } from '../utils/apis';
+
+export function ArticleActions({ currentArticle, setCurrentArticle }) {
+  const { article_id, comment_count, votes } = currentArticle;
+  const [userAction, setUserAction] = useState({
+    hasCommented: false,
+    hasVoted: false,
+  });
+  const [err, setErr] = useState(null);
+
+
+
+  const changeVoteCount = () => {
+    const voteChange = userAction.hasVoted ? -1 : 1;
+    const newHasVoted = !userAction.hasVoted;
+
+    setCurrentArticle((prevArticle) => {
+      return { ...prevArticle, votes: votes + voteChange };
+    });
+    setUserAction({ ...userAction, hasVoted: newHasVoted });
```

Comment:
I'm finding it tricky to follow your logic here but I don't think we need newHasVoted?